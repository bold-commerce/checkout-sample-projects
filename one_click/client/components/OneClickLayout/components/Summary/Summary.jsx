import React from 'react';
import SummaryLine from './SummaryLine';
import SummaryItem from './SummaryItem';
import { Price } from '@boldcommerce/stacks-ui/lib';
import { BackButton } from '../BackButton';
import { useShippingLines, useDiscount, useCheckoutStore } from '@boldcommerce/checkout-react-components';
import './Summary.scss';

const Summary = () => {
    const { data: shipping } = useShippingLines();
    const { data: discount, removeDiscount } = useDiscount();
    const { state } = useCheckoutStore();
    const orderTotals =  state.orderTotals;
    const taxes = state.applicationState.taxes;

    let shippingItems = null;
    let discountItems = null;
    let taxItems = null;
    
    if ( shipping.selectedShippingDescription && shipping.selectedShippingAmount){
        shippingItems = <SummaryItem
                            title={shipping.selectedShippingDescription}
                            value={shipping.selectedShippingAmount}
                        />;
    }

    if (discount.discountApplied) {
        discountItems = <SummaryItem 
                            title={discount.discountCode}
                            value={(discount.discountTotal * -1)}
                            removeItem={removeDiscount}
                        />;
    }

    if (taxes.length > 0) {
        taxItems = taxes.map( (taxline) => {
            return(                
                <SummaryItem
                    key={taxline.name}
                    title={taxline.name}
                    value={taxline.value}
                />
            )
        })
    }

    return(
        <div className="Summary">
            <h1 className="Section__Title">Summary</h1>
            <BackButton />
            <section className="Summary__OrderSummary">
                <div className="Summary__Lines SummaryBlock" data-allow-multiple id="accordianGroup">
                    <SummaryLine
                        title="Subtotal"
                        value={orderTotals.subTotal}
                    />
                    <SummaryLine
                        title="Shipping"
                        value={shipping.shippingLines.selectedShippingAmount}
                        items={shippingItems}
                    />
                    
                    <SummaryLine
                        title="Discount"
                        value={( discount.discountApplied ? discount.discountTotal * -1 : undefined )}
                        items={discountItems}
                    />
                    <SummaryLine
                        title="Taxes"
                        value={( orderTotals.taxesTotal !== 0 ? orderTotals.taxesTotal : undefined )}
                        items={taxItems}
                    />
                </div>                
                <div className="Summary__Total SummaryBlock">
                    <div>Total</div>
                    <Price className="summary-total-price" amount={orderTotals.total} />
                </div>
            </section>
        </div>
    );
};

export default Summary;