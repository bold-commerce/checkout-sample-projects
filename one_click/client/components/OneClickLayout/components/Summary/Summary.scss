.Summary__ {
    &Lines.SummaryBlock,
    &Total.SummaryBlock {
        padding: 16px 0;
    }

    &Total {
        color: rgba(0, 0, 0, .9);
        line-height: 20px;
        font-size: 18px;
        display: flex;
        justify-content: space-between;
        font-weight: 700;

        & > div {
            margin-left: 24px;
        }
    }
}

.Summary > .CheckoutButton__Container {
    border-top: none;
}

.summary-line {
    flex-direction: column;
    padding-top: 8px;
    gap: 8px;
    font-size: 14px;
    line-height: 20px;
    color: rgba(0, 0, 0, .65);
    
    &:first-child {
        padding-top: 0px;
    }

    &-child,
    &-title,
    &-head {
        display: flex;
        justify-content: space-between;
    }

    &-icon {
        width: 20px;
        height: 20px;
        margin-right: 4px;
    }

    &-content {
        margin-left: 24px;
    }

    &-child {
        padding-top: 8px;
    }
}

.summary-child {

    &-remove {
        margin-left: 8px;
    }

    &-title {
        display: flex;
    }
}

button.summary-btn,
button.summary-child-remove {
    border: none;
    display: inherit;
    justify-content: inherit;
    text-decoration: none;
    font-size: inherit;
    line-height: inherit;
    color: inherit;
    background: transparent;
    width: 100%;
    cursor: pointer;
}

.stx-price__amount {
    font-size: 14px;
    font-weight: normal;
    line-height: 20px;
    color: rgba(0, 0, 0, .65);
}

.summary-total-price > .stx-price__amount {
    color: rgba(0, 0, 0, .9);
    line-height: 20px;
    font-size: 18px;
    font-weight: 700;
}