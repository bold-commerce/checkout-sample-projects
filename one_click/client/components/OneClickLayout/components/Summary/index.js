export { default as Summary } from './Summary';
export { default as SummaryCondensed } from './SummaryCondensed';