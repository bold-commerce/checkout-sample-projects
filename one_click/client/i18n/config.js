import i18n from 'i18next';
import enTranslation from './en/translation.json';
import { initReactI18next } from 'react-i18next';

i18n
  .use(initReactI18next)
  .init({
    resources: {
      en: {
        translation: enTranslation
      }
    },
    lng: 'en',
    fallbackLng: 'en',
    interpolation: {
      escapeValue: false, 
    }
  });

export default i18n;