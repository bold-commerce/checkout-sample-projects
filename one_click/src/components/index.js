export * from './Address';
export * from './BackButton';
export * from './Billing';
export * from './BillingAddress';
export * from './Card';
export * from './CheckoutButton';
export * from './CheckoutForm';
export * from './Confirmation';
export * from './CustomerInfo';
export * from './Discount';
export * from './EmptyState';
export * from './Header';
export * from './Icons';
export * from './Index';
export * from './InventoryItem';
export * from './LineItems';
export * from './LoadingState';
export * from './Payment';
export * from '../pages/Processing';
export * from './Product';
export * from './RadioItem';
export * from './Shipping';
export * from './ShippingLines';
export * from './Summary';
