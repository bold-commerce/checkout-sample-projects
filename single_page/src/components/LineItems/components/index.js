export { default as LineItem } from './LineItem';
export { default as Product } from './Product';
