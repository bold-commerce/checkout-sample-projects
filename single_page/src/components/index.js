export * from './Address';
export * from './BillingAddress';
export * from './CheckoutButton';
export * from './CheckoutSection';
export * from './Customer';
export * from './DiscountForm';
export * from './EmptyState';
export * from './Header';
export * from './LineItems';
export * from './LoadingState';
export * from './OrderErrors';
export * from './OrderSummary';
export * from './PaymentMethod';
export * from './ShippingAddress';
export * from './ShippingLines';
