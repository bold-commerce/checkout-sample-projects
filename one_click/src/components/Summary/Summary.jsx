import React, { useCallback } from 'react';
import { useShippingLines, useDiscount, useCheckoutStore } from '@boldcommerce/checkout-react-components';
import { Price } from '@boldcommerce/stacks-ui';
import SummaryLine from './SummaryLine';
import SummaryItem from './SummaryItem';
import { BackButton } from '../BackButton';
import { Header } from '../Header';
import classNames from 'classnames';
import { CheckoutButton } from '../CheckoutButton';
import { useAnalytics, useErrorLogging } from '../../hooks';
import './Summary.scss';
import { useTranslation } from 'react-i18next';

const Summary = ({ section, onSectionChange }, ref) => {
  const { data: shipping } = useShippingLines();
  const { data: discount, removeDiscount } = useDiscount();
  const { state } = useCheckoutStore();
  const orderTotals =  state.orderTotals;
  const taxes = state.applicationState.taxes;
  const trackEvent = useAnalytics();
  const logError = useErrorLogging();
  const show = section === 'summary' || section === 'summaryB'
  const backLocation = section === 'summaryB' ? 'billing' : '/';
  const { t } = useTranslation();

  const handleRemoveDiscount = useCallback(async () => {
    try {
      await removeDiscount(discount.discountCode);
      trackEvent('remove_discount_code');
    } catch(e) {
      logError('discount_code', e);
    }
  }, [discount.discountCode]);

  let shippingItems = null;
  let discountItems = null;
  let taxItems = null;
  
  if ( shipping.selectedShippingDescription && shipping.selectedShippingAmount){
  shippingItems = <SummaryItem
                    title={shipping.selectedShippingDescription}
                    value={shipping.selectedShippingAmount}
                  />;
  }

  if (discount.discountApplied) {
    discountItems = <SummaryItem 
                      title={discount.discountCode}
                      value={(discount.discountTotal * -1)}
                      removeItem={handleRemoveDiscount}
                    />;
  }

  if (taxes.length > 0) {
    taxItems = taxes.map( (taxline) => {
      return(                
        <SummaryItem
          key={taxline.name}
          title={taxline.name}
          value={taxline.value}
        />
      );
    });
  }

  return(
    <div ref={ref} className={classNames('Sidebar Summary', show ? 'Sidebar--Show' : 'Sidebar--Hide')}>
      <Header title={t('summary.title')} />
      <BackButton onClick={() => onSectionChange(backLocation)} />
      <section className="Summary__OrderSummary">
        <div className="Summary__Lines SummaryBlock" data-allow-multiple id="accordianGroup">
          <SummaryLine
            title={t('summary.subtotal')}
            value={orderTotals.subTotal}
          />
          <SummaryLine
            title={t('shipping.title')}
            value={shipping.selectedShippingAmount}
            items={shippingItems}
          />
          <SummaryLine
            title={t('discount.title')}
            value={( discount.discountApplied ? discount.discountTotal * -1 : undefined )}
            items={discountItems}
          />
          <SummaryLine
            title={t('summary.taxes')}
            value={( orderTotals.taxesTotal !== 0 ? orderTotals.taxesTotal : undefined )}
            items={taxItems}
          />
        </div>                
        <div className="Summary__Total SummaryBlock">
          <div>{t('summary.total')}</div>
          <Price className="summary-total-price" amount={orderTotals.total} />
        </div>
      </section>
      <CheckoutButton className="CheckoutButton CheckoutButton__Mobile" />
    </div>
  );
};

const SummaryForwardedRef = React.forwardRef(Summary);

export default SummaryForwardedRef;