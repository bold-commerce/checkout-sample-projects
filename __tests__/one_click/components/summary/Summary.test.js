import React from 'react';
import { render } from '@testing-library/react';
import { Summary } from '../../../../one_click/src/components/Summary'
import { exampleLineItems as MOCKexampleLineItems } from '../../../utils/lineItemHelpers';
import { testApplicationState as MOCKapplicationState } from '../../../utils/applicationStateHelper';
import { exampleShippingState as MOCKexampleShippingState } from '../../../utils/shippingLinesHelper';
import '../../../../one_click/src/i18n/config';

jest.mock('@boldcommerce/checkout-react-components', () => ({
    ...jest.requireActual('@boldcommerce/checkout-react-components'),
    useCheckoutStore: () => ({
        state: {
            orderInfo: {
                billingSameAsShipping:true,
                orderStatus: ""
            },
            errors: { order: null },
            loadingStatus: { isLoading: false },
            applicationState: MOCKapplicationState,
            orderTotals: {
                taxesTotal: 1200,
                subTotal: 23000,
                total: 24200
            }
        }
    }),
    useShippingLines: () => ({
        data: MOCKexampleShippingState
    }),
    useDiscount: () => ({
        data: '?',
        applyDiscount : (() => {})
    }),
    usePaymentIframe: () => ({
        data: {
            url: 'test.url',
            loadingStatus: 'fulfilled',
            paymentIframeOnLoaded: (() => {})
        }
    }),
    useLineItems: () => ({
        data: MOCKexampleLineItems
    })
})).mock('react', () => ({
    ...jest.requireActual('react'),
    useContext: () => ({
        setShowCheckout: (() => {})
    })
})).mock('react-router', () => ({
    ...jest.requireActual('react-router'),
    useLocation: () => ({
        pathname: '/' 
    }),
    useNavigate: () => ({
        navigate: (() => {})
    })
}));

describe('Summary', () => {
    test('renders Summary component', () => {
        const { asFragment } = render( <Summary /> );

        expect(asFragment()).toMatchSnapshot();
    });
});
