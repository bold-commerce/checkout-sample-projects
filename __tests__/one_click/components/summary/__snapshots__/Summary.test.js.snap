// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary renders Summary component 1`] = `
<DocumentFragment>
  <div
    class="Sidebar Summary Sidebar--Hide"
  >
    <div
      class="Header"
    >
      <h1
        class="Header__Title"
      >
        Summary
      </h1>
      <button
        aria-label="Close"
        class="Header__Times"
      >
        <svg
          aria-labelledby="button"
          fill="none"
          height="20"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            opacity="0.8"
          >
            <path
              clip-rule="evenodd"
              d="M17.7071 7.70706L7.70712 17.7071L6.29291 16.2928L16.2929 6.29285L17.7071 7.70706Z"
              fill="black"
              fill-rule="evenodd"
            />
            <path
              clip-rule="evenodd"
              d="M16.2929 17.7071L6.29291 7.70706L7.70712 6.29285L17.7071 16.2928L16.2929 17.7071Z"
              fill="black"
              fill-rule="evenodd"
            />
          </g>
        </svg>
      </button>
    </div>
    <div>
      <button
        class="back-btn"
        type="button"
      >
        Back
      </button>
    </div>
    <section
      class="Summary__OrderSummary"
    >
      <div
        class="Summary__Lines SummaryBlock"
        data-allow-multiple="true"
        id="accordianGroup"
      >
        <div
          class="summary-line"
        >
          <div
            class="summary-line-head"
          >
            <div
              class="summary-line-title"
            >
              <div
                class="summary-line-icon"
              />
              Subtotal
            </div>
            <div
              class="stx-price summary-line-value stx-price--alight-center"
            >
              <p
                aria-label="Product price"
                class="stx-price__amount"
              >
                $230.00 CAD
              </p>
            </div>
          </div>
        </div>
        <div
          class="summary-line"
        >
          <div
            class="summary-line-head"
          >
            <button
              class="summary-btn"
              id="summary-line-Shipping"
            >
              <div
                class="Accordion-title summary-line-title"
              >
                <div
                  class="summary-line-icon"
                  style="transform: rotate(0deg);"
                >
                  <svg
                    fill="none"
                    height="20"
                    viewBox="0 0 24 24"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      opacity="0.8"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M9 5.91943L16.6008 12.0001L9 18.0807V5.91943ZM11 10.0807V13.9194L13.3992 12.0001L11 10.0807Z"
                        fill="black"
                        fill-rule="evenodd"
                      />
                    </g>
                  </svg>
                </div>
                Shipping
              </div>
              <div
                class="stx-price summary-line-value stx-price--alight-center"
              >
                <p
                  aria-label="Product price"
                  class="stx-price__amount"
                >
                  $20.00 CAD
                </p>
              </div>
            </button>
          </div>
          <div
            aria-labelledby="summary-line-Shipping"
            class="summary-line-content"
            hidden=""
            id="summary-line-Shipping-div"
            role="region"
          >
            <div
              class="summary-line-child"
            >
              <div
                class="summary-child-title"
              >
                Standard Shipping
              </div>
              <div
                class="stx-price summary-child-value stx-price--alight-center"
              >
                <p
                  aria-label="Product price"
                  class="stx-price__amount"
                >
                  $20.00 CAD
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="summary-line"
        >
          <div
            class="summary-line-head"
          >
            <button
              class="summary-btn"
              id="summary-line-Discount"
            >
              <div
                class="Accordion-title summary-line-title"
              >
                <div
                  class="summary-line-icon"
                  style="transform: rotate(0deg);"
                >
                  <svg
                    fill="none"
                    height="20"
                    viewBox="0 0 24 24"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      opacity="0.8"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M9 5.91943L16.6008 12.0001L9 18.0807V5.91943ZM11 10.0807V13.9194L13.3992 12.0001L11 10.0807Z"
                        fill="black"
                        fill-rule="evenodd"
                      />
                    </g>
                  </svg>
                </div>
                Discount
              </div>
              <div
                class="stx-price summary-line-value stx-price--alight-center"
              >
                <p
                  aria-label="Product price"
                  class="stx-price__amount"
                >
                  $-13.37 CAD
                </p>
              </div>
            </button>
          </div>
          <div
            aria-labelledby="summary-line-Discount"
            class="summary-line-content"
            hidden=""
            id="summary-line-Discount-div"
            role="region"
          >
            <div
              class="summary-line-child"
            >
              <div
                class="summary-child-title"
              >
                discount_code
                <button
                  aria-label="remove item"
                  class="summary-child-remove"
                >
                  <svg
                    aria-labelledby="button"
                    fill="none"
                    height="20"
                    viewBox="0 0 24 24"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      opacity="0.8"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M17.7071 7.70706L7.70712 17.7071L6.29291 16.2928L16.2929 6.29285L17.7071 7.70706Z"
                        fill="black"
                        fill-rule="evenodd"
                      />
                      <path
                        clip-rule="evenodd"
                        d="M16.2929 17.7071L6.29291 7.70706L7.70712 6.29285L17.7071 16.2928L16.2929 17.7071Z"
                        fill="black"
                        fill-rule="evenodd"
                      />
                      <path
                        clip-rule="evenodd"
                        d="M12 3C7.02944 3 3 7.02944 3 12C3 16.9706 7.02944 21 12 21C16.9706 21 21 16.9706 21 12C21 7.02944 16.9706 3 12 3ZM1 12C1 5.92487 5.92487 1 12 1C18.0751 1 23 5.92487 23 12C23 18.0751 18.0751 23 12 23C5.92487 23 1 18.0751 1 12Z"
                        fill="black"
                        fill-rule="evenodd"
                      />
                    </g>
                  </svg>
                </button>
              </div>
              <div
                class="stx-price summary-child-value stx-price--alight-center"
              >
                <p
                  aria-label="Product price"
                  class="stx-price__amount"
                >
                  $-13.37 CAD
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="summary-line"
        >
          <div
            class="summary-line-head"
          >
            <button
              class="summary-btn"
              id="summary-line-Taxes"
            >
              <div
                class="Accordion-title summary-line-title"
              >
                <div
                  class="summary-line-icon"
                  style="transform: rotate(0deg);"
                >
                  <svg
                    fill="none"
                    height="20"
                    viewBox="0 0 24 24"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      opacity="0.8"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M9 5.91943L16.6008 12.0001L9 18.0807V5.91943ZM11 10.0807V13.9194L13.3992 12.0001L11 10.0807Z"
                        fill="black"
                        fill-rule="evenodd"
                      />
                    </g>
                  </svg>
                </div>
                Taxes
              </div>
              <div
                class="stx-price summary-line-value stx-price--alight-center"
              >
                <p
                  aria-label="Product price"
                  class="stx-price__amount"
                >
                  $12.00 CAD
                </p>
              </div>
            </button>
          </div>
          <div
            aria-labelledby="summary-line-Taxes"
            class="summary-line-content"
            hidden=""
            id="summary-line-Taxes-div"
            role="region"
          >
            <div
              class="summary-line-child"
            >
              <div
                class="summary-child-title"
              >
                GST
              </div>
              <div
                class="stx-price summary-child-value stx-price--alight-center"
              >
                <p
                  aria-label="Product price"
                  class="stx-price__amount"
                >
                  $11.25 CAD
                </p>
              </div>
            </div>
            <div
              class="summary-line-child"
            >
              <div
                class="summary-child-title"
              >
                PST
              </div>
              <div
                class="stx-price summary-child-value stx-price--alight-center"
              >
                <p
                  aria-label="Product price"
                  class="stx-price__amount"
                >
                  $15.75 CAD
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="Summary__Total SummaryBlock"
      >
        <div>
          Total
        </div>
        <div
          class="stx-price summary-total-price stx-price--alight-center"
        >
          <p
            aria-label="Product price"
            class="stx-price__amount"
          >
            $242.00 CAD
          </p>
        </div>
      </div>
    </section>
    <div
      class="CheckoutButton__Container"
    >
      <button
        aria-busy="false"
        aria-disabled="false"
        class="stx-button CheckoutButton CheckoutButton__Mobile"
      >
        <span
          class="stx-button__text"
        >
          Complete Order
        </span>
        <span
          class="stx-loading-spinner stx-button__spinner"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
